airbyte-cdk/python/airbyte_cdk/sources/concurrent_source/concurrent_source_adapter.py

```diff
@@ -41,7 +44,7 @@ def read(
         if abstract_streams:
             yield from self._concurrent_source.read(abstract_streams)
         if configured_catalog_for_regular_streams.streams:
-            yield from super().read(logger, config, configured_catalog_for_regular_streams, state)
+            yield from super().read(logger, config, configured_catalog_for_regular_streams, state)  # type: ignore[arg-type]
```

Comment:
can you add a comment explaining why this is fine?

Reply:
added state transformation to List[AirbyteStateMessage] data type when we use AbstractSource read method